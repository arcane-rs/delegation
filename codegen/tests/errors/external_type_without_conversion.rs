use delegation::{__macros::Either, delegate};

#[delegate]
trait AsStr {
    fn as_str(&self) -> &str;
}

impl AsStr for String {
    fn as_str(&self) -> &str {
        self
    }
}

#[delegate(derive(AsStr))]
enum EitherDef {
    Left(String),
    Right(String),
}

#[delegate(derive(AsStr))]
struct EitherString(#[delegate(as = EitherDef)] Either<String, String>);

fn main() {
    unreachable!()
}
