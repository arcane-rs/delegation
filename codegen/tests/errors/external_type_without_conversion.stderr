error[E0277]: the trait bound `EitherDef: From<Either<String, String>>` is not satisfied
  --> tests/errors/external_type_without_conversion.rs:20:1
   |
20 | #[delegate(derive(AsStr))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<Either<String, String>>` is not implemented for `EitherDef`
   |
   = note: this error originates in the attribute macro `delegate` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `&EitherDef: From<&Either<String, String>>` is not satisfied
  --> tests/errors/external_type_without_conversion.rs:20:1
   |
20 | #[delegate(derive(AsStr))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<&Either<String, String>>` is not implemented for `&EitherDef`
   |
   = note: this error originates in the attribute macro `delegate` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `&mut EitherDef: From<&mut Either<String, String>>` is not satisfied
  --> tests/errors/external_type_without_conversion.rs:20:1
   |
20 | #[delegate(derive(AsStr))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<&mut Either<String, String>>` is not implemented for `&mut EitherDef`
   |
   = note: this error originates in the attribute macro `delegate` (in Nightly builds, run with -Z macro-backtrace for more info)
